File: frontend/demo/component/avatar/avatar-group-basic.ts
import '../../init'; // hidden-full-source-line

import { html, LitElement, customElement, internalProperty } from 'lit-element';
import '@vaadin/vaadin-avatar/vaadin-avatar-group';
import { applyTheme } from 'generated/theme';
import { getPeople } from '../../domain/DataService';
import Person from '../../../generated/com/vaadin/demo/domain/Person';

@customElement('avatar-group-basic')
export class Example extends LitElement {
  constructor() {
    super();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(this.shadowRoot);
  }

  @internalProperty()
  private items: Person[] = [];

  async firstUpdated() {
    this.items = await getPeople(3);
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-avatar-group
        .items=${this.items.map(person => {
          return {
            name: `${person.firstName} ${person.lastName}`
          };
        })}
      >
      </vaadin-avatar-group>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/avatar/AvatarGroupBasic.java
package com.vaadin.demo.component.avatar;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;
import com.vaadin.demo.domain.Person;
import com.vaadin.flow.component.avatar.AvatarGroup;
import com.vaadin.flow.component.avatar.AvatarGroup.AvatarGroupItem;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import java.util.List;

@Route("avatar-group-basic")
public class AvatarGroupBasic extends Div {

  private List<Person> people = DataService.getPeople(3);

  public AvatarGroupBasic() {
    // tag::snippet[]
    AvatarGroup avatarGroup = new AvatarGroup();

    for (Person person : people) {
      String name = person.getFirstName() + " " + person.getLastName();
      AvatarGroupItem avatar = new AvatarGroupItem(name);
      avatarGroup.add(avatar);
    }
    // end::snippet[]

    add(avatarGroup);
  }

  public static class Exporter extends DemoExporter<AvatarGroupBasic> { // hidden-source-line
  } // hidden-source-line
}
